addons/stock_account/models/account_move.py

```diff
@@ -6,7 +6,7 @@
 class AccountMove(models.Model):
     _inherit = 'account.move'
 
-    stock_move_id = fields.Many2one('stock.move', string='Stock Move', index=True)
+    stock_move_id = fields.Many2one('stock.move', string='Stock Move', index='not null')
```

Comment:
Again this looks tuned for non trading companies but in a stock module. The not null fraction in a trading database is typically under 30%.